pkg/credentials/credentials.go

```diff
@@ -81,11 +86,18 @@ type Expiry struct {
 // Using a window is helpful to trigger credentials to expire sooner than
 // the expiration time given to ensure no requests are made with expired
 // tokens.
+//
+// window argument is deprecated in August 2021
 func (e *Expiry) SetExpiration(expiration time.Time, window time.Duration) {
-	e.expiration = expiration
-	if window > 0 {
-		e.expiration = e.expiration.Add(-window)
+	if e.CurrentTime == nil {
+		e.CurrentTime = time.Now
+	}
+	expireIn := expiration.Sub(e.CurrentTime())
```

Comment:
```suggestion
	expireIn := time.Until(expiration)
```

Reply:
I am actually not sure why we have CurrentTime in the first place, that's why I am using it.

Reply:
@vadmeste Yeah, lets keep it.

Reply:
> @vadmeste Yeah, lets keep it.

It looks like it is used for testing.